presto-main/src/main/java/com/facebook/presto/sql/relational/FunctionResolution.java

```diff
@@ -303,4 +304,16 @@ public FunctionHandle minFunction(Type valueType)
     {
         return functionAndTypeManager.lookupFunction("min", fromTypes(valueType));
     }
+
+    public static boolean isBuiltInEqualFunction(FunctionHandle functionHandle)
```

Comment:
You can just call this `isEqualFunction`. Everything in this class already assumes "builtin". Also please use `FunctionAndTypeManager` to resolve function rather than adding specific checks. You can pass in a reference to `FunctionResolution` class when creating `LogicalPropertiesProviderImpl` from `PlanOptimizers`.

Reply:
Done.